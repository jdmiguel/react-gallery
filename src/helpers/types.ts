export type ThumbData = {
  id: number;
  src: string;
  title: string;
}

export type DetailData = {
  id: number;
  src: string;
  title: string;
  views: number;
  downloads: number;
  favorites: number;
  likes: number;
}

export type ImageExtendedData = {
  id: number;
  pageURL: string;
  type: string;
  tags: string;
  previewURL: string;
  previewWidth: number;
  previewHeight: number;
  webformatURL: string;
  webformatWidth: number;
  webformatHeight: number;
  largeImageURL: string;
  fullHDURL: string;
  imageURL: string;
  imageWidth: number;
  imageHeight: number;
  imageSize: number;
  views: number;
  downloads: number;
  favorites: number;
  likes: number;
  comments: number;
  user_id: number;
  user: string;
  userImageURL: string;
}