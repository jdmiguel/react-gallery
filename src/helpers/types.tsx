export type Image = {
  src: string;
}