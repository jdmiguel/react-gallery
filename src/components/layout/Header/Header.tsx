import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

import Logo from '../../core/Logo/Logo';

const StyledHeader = styled.div`
  width: 100%;
  max-width: 1100px;
  height: 80px;
  display: flex;
  justify-content: space-around;
  align-items: center;
  @media only screen and (min-width: 1200px) {
    justify-content: space-between;
  }
`;

const StyledHeaderClaim = styled.h2`
  font-size: 1.3rem;
  font-weight: 400;
  color: ${(props) => props.theme.palette.LIGHT_MIN};
  display: none;
  @media only screen and (min-width: 768px) {
    display: block;
  }
  @media only screen and (min-width: 992px) {
    font-size: 1.5rem;
  }
`;

const Header: React.FC = () => (
    <StyledHeader>
      <Link to="/">
        <Logo />
      </Link>
      <StyledHeaderClaim>Find out all the images you can imagine!</StyledHeaderClaim>
    </StyledHeader>
);

export default Header;