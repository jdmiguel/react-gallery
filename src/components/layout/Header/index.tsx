import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

import Logo from '../../core/Logo';

import BgHeader from './gallery.jpg';

const StyledHeader = styled.header`
  width: 100%;
  height: 120px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-image: url(${BgHeader});
  background-size: cover;
  background-color: ${({theme}) => theme.palette.PRIMARY_DARK};
  background-position: 70%;
  background-blend-mode: multiply;
`;

const StyledHeaderClaim = styled.h2`
  margin-top: 10px;
  font-size: 1rem;
  font-weight: 400;
  color:  ${({theme}) => theme.palette.LIGHT_MIN};
  @media only screen and (min-width: 768px) {
    font-size: 1.5rem;
  }
`;

const Header: React.FC = () => (
    <StyledHeader>
      <Link to="/">
        <Logo />
      </Link>
      <StyledHeaderClaim>Find out all the images you can imagine!</StyledHeaderClaim>
    </StyledHeader>
);

export default Header;