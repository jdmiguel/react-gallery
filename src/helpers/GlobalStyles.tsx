import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  body {
    margin: 0;
    padding: 0;
    font-family: 'Josefin Sans', sans-serif;
    font-size: 15px;
    color: #2f3235;
    background-color: #F3F3F3;
  }

  h1, h2, h3, h4, h5, h6, p{
    margin: 0;
  }

  #root{
    overflow: auto;
  }
`;