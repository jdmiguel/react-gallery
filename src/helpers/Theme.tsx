import React, { ReactNode } from 'react';
import { DefaultTheme, ThemeProvider } from 'styled-components';

export const theme: DefaultTheme = {
  palette: {
    PRIMARY: '#A66885',
    PRIMARY_DARK: '#734359',
    PRIMARY_LIGHT: '#F2C1AE',
    LIGHT_MAX: '#E5EAED',
    LIGHT_MIN: '#C8C8C8',
    GREY: '#E5EAED',
    DARK_MAX: '#2f3235',
    DARK_MIN: '#798186',
  },
};

export const renderWithTheme = (component: ReactNode) => (
  <ThemeProvider theme={theme}>{component}</ThemeProvider>
);
