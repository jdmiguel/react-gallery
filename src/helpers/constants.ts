// API
export const API_PIXABAY = "https://pixabay.com/api/";
export const API_KEY = "6473511-0417f2cad683f1bee54cafe15";

// IMAGES
export const DEFAULT_THUMBS = [
  {
    id: 0,
    src: '',
    title: ''
  },
];

// IMAGE LOADING
export const INITIAL_PAGE = 1;
export const MAX_PAGES_ALLOWED = 20;
export const INITIAL_MAX_IMAGES_LOADED = 40;
export const MAX_IMAGES_LOADED_BY_SCROLL = 20;
export const MIN_FOOTER_VIEWS = 3;
export const OBSERVER_OPTIONS = {
  root: document.querySelector('root'),
  rootMargin: '15%',
  threshold: 0,
};