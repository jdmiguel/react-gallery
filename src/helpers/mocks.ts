export const THUMB_PROPS_MOCKED = {
  id: 3063284,
  src: "https://pixabay.com/get/55e0d340485aa814f1dc84609629327f113dd7e05b4c704f752f78d79744c15e_640.jpg",
  title: "flower"
};
export const BUTTON_TEXT_MOCKED = "default button";
export const Pill_TEXT_MOCKED = "default pill";