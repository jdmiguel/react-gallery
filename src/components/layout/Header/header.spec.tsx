import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';

import Header from '.';

import { renderWithTheme } from '../../../helpers/Theme';

describe('Component: Header', () => {
  it('should render', () => {
    const { container } = render(
      renderWithTheme( 
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      )
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
