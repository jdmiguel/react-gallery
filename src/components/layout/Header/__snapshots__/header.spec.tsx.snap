// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Header should render 1`] = `
.c1 {
  font-size: 1.8rem;
  font-weight: 700;
  color: #F2C1AE;
}

.c0 {
  width: 100%;
  height: 120px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-image: url(gallery.jpg);
  background-size: cover;
  background-color: #734359;
  background-position: 70%;
  background-blend-mode: multiply;
}

.c2 {
  margin-top: 10px;
  font-size: 1rem;
  font-weight: 400;
  color: #C8C8C8;
}

@media only screen and (min-width:768px) {
  .c1 {
    font-size: 2.5rem;
  }
}

@media only screen and (min-width:768px) {
  .c2 {
    font-size: 1.5rem;
  }
}

<header
  class="c0"
>
  <a
    href="/"
  >
    <h1
      class="c1"
    >
      React Gallery
    </h1>
  </a>
  <h2
    class="c2"
  >
    Find out all the images you can imagine!
  </h2>
</header>
`;
